resalloc_ibm_cloud/ibm_cloud_vm.py

```diff
@@ -339,6 +339,11 @@ def main():
         opts.instance_name = name
         opts.instance = "production" if "-prod-" in name else "devel"
 
+    maxlen = 56
+    if len(name) > maxlen:
+        log.error("Name %s is too long, max %s characters", name, maxlen)
+        sys.exit(1)
```

Comment:
Could we do something else thank error? I mean something like shrinking the name?

Reply:
Thank you for looking.  Could be?  But it is reproducible with:

```
RESALLOC_NAME=s390x_us_east /var/lib/resallocserver/resalloc_provision/ibm-cloud-vm create "copr_ibm_cloud_s390x_washington_prod_13290257_20231024_201117"
...
ibm_cloud_sdk_core.api_exception.ApiException: Error: Expected only one oneOf fields to be set: got 0, Code: 400
```

While it is just ok with one byte dropped:

```
RESALLOC_NAME=s390x_us_east /var/lib/resallocserver/resalloc_provision/ibm-cloud-vm create "copr_ibm_cloud_s390x_washington_prod_13290257_20231024_20111"
...
```

I'd swear I was able to reproduce it with `copr-ibm-cloud-s390x-washington-dev-02817070-20231024-10x` string while bisecting.  See differences:

```
copr_ibm_cloud_s390x_washington_prod_13290257_20231024_201117  fails (61 bytes)
copr_ibm_cloud_s390x_washington_prod_13290257_20231024_20111   works (60 bytes)
copr-ibm-cloud-s390x-washington-dev-02817070-20231024-10x      works now, failed before
```

Reply:
Ok, now I think the problem was the name of the volume, which is basically instance name + suffix.  PTAL